diesel/src/query_builder/select_statement/dsl_impls.rs

```diff
@@ -415,17 +313,17 @@ impl<F, S, D, W, O, L, Of, G, LM> SkipLockedDsl
             self.limit,
             self.offset,
             self.group_by,
-            LockingClause::new(self.locking.lock_mode, SkipLocked),
+            LockingClause::new(lock, NoModifier),
         )
     }
 }
 
-impl<F, S, D, W, O, L, Of, G, LM> NoWaitDsl
-    for SelectStatement<F, S, D, W, O, L, Of, G, LockingClause<LM, NoModifier>>
+impl<F, S, D, W, O, L, Of, G, LC, LM, Modifier> ModifyLockDsl<Modifier>
```

Comment:
Was it intentional to allow setting the modifier multiple times here?

eg. `for_update().skip_locked().no_wait()`

Reply:
Yes. We generally allow overriding clauses later. This should be no different. I'm planning on a followup PR to allow overriding the rest of the lock clause as well